Helpers/ShareOfCheckout/ShareOfCheckoutHelper.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+namespace Alma\MonthlyPayments\Helpers\ShareOfCheckout;
+
+use Alma\API\RequestError;
+use Alma\MonthlyPayments\Helpers\AlmaClient;
+use Alma\MonthlyPayments\Helpers\ConfigHelper;
+use Alma\MonthlyPayments\Helpers\Logger;
+use InvalidArgumentException;
+use Magento\Framework\App\Config\Storage\WriterInterface;
+use Magento\Framework\App\Helper\AbstractHelper;
+use Magento\Framework\App\Helper\Context;
+use Magento\Store\Model\ScopeInterface;
+
+class ShareOfCheckoutHelper extends AbstractHelper
+{
+
+    const SHARED_ORDER_STATES = ['processing', 'complete'];
+    const SHARE_CHECKOUT_ENABLE_KEY = 'share_checkout_enable';
+    const SHARE_CHECKOUT_DATE_KEY = 'share_checkout_date';
+
+
+    /**
+     * @var Logger
+     */
+    private $logger;
+    /**
+     * @var AlmaClient
+     */
+    private $almaClient;
+    /**
+     * @var WriterInterface
+     */
+    private $configWriter;
+    /**
+     * @var PayloadBuilder
+     */
+    private $payloadBuilder;
+    /**
+     * @var DateHelper
+     */
+    private $dateHelper;
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @param Logger $logger
+     * @param AlmaClient $almaClient
+     * @param Context $context
+     * @param WriterInterface $configWriter
+     * @param PayloadBuilder $payloadBuilder
+     * @param OrderHelper $orderHelper
+     * @param DateHelper $dateHelper
+     */
+    public function __construct(
+        Context $context,
+        Logger $logger,
+        AlmaClient $almaClient,
+        WriterInterface $configWriter,
+        PayloadBuilder $payloadBuilder,
+        OrderHelper $orderHelper,
+        DateHelper $dateHelper
+    ) {
+        parent::__construct($context);
+        $this->logger = $logger;
+        $this->almaClient = $almaClient->getDefaultClient();
+        $this->configWriter = $configWriter;
+        $this->payloadBuilder = $payloadBuilder;
+        $this->orderHelper = $orderHelper;
+        $this->dateHelper = $dateHelper;
+    }
+
+    /**
+     * @return bool
+     */
+    public function shareOfCheckoutIsEnabled(): bool
+    {
+        return $this->scopeConfig->getValue(
+            ConfigHelper::XML_PATH_PAYMENT . '/' . ConfigHelper::XML_PATH_METHODE . '/' . self::SHARE_CHECKOUT_ENABLE_KEY,
+            ScopeInterface::SCOPE_STORE
+        );
+    }
+
+    /**
+     * @param string $date
+     * @return void
+     * @throws RequestError
+     */
+    public function shareDay(string $date): void
+    {
+        if (!$this->almaClient) {
```

Comment:
shouldn't we check this on the constructor instead of checking this in every function ? Is there an issue with the client ?

Reply:
Done